spring-cloud-gcp-starters/spring-cloud-gcp-starter-core/src/main/java/org/springframework/cloud/gcp/core/autoconfig/GCPContextAutoConfiguration.java

```diff
@@ -2,38 +2,39 @@
 
 import java.io.FileInputStream;
 
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
-import org.springframework.context.EnvironmentAware;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.gcp.core.GCPProperties;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
-import org.springframework.core.env.Environment;
 import org.springframework.core.io.ResourceLoader;
 import org.springframework.util.StringUtils;
 
 import com.google.auth.oauth2.GoogleCredentials;
 
 /**
  * @author Vinicius Carvalho
+ *
+ * Base starter for Google Cloud Projects. Provide defaults for {@link GoogleCredentials}.
+ * Binds properties from {@link GCPProperties}
+ *
  */
 @Configuration
 @ConditionalOnClass(GoogleCredentials.class)
-public class CredentialsAutoConfiguration implements EnvironmentAware {
+@EnableConfigurationProperties(GCPProperties.class)
+public class GCPContextAutoConfiguration {
 
-	private static final String AUTH_LOCATION_KEY = "cloud.gcp.auth.location";
 
-	private Environment environment;
-
-	@Override
-	public void setEnvironment(Environment environment) {
-		this.environment = environment;
-	}
+	@Autowired
+	private GCPProperties gcpProperties;
 
 	@Bean
 	public GoogleCredentials googleCredentials(ResourceLoader resourceLoader)
 			throws Exception {
-		if (!StringUtils.isEmpty(this.environment.getProperty(AUTH_LOCATION_KEY))) {
+		if (!StringUtils.isEmpty(this.gcpProperties.getJsonKeyFile())) {
 			return GoogleCredentials.fromStream(new FileInputStream(resourceLoader
-					.getResource(this.environment.getProperty(AUTH_LOCATION_KEY))
+					.getResource(this.gcpProperties.getJsonKeyFile())
 					.getFile()));
 		}
 		else {
```

Comment:
I think @joaoandremartins was referring to the fact that the `if` block above had a `return`, not just the fact that there's one line in the `else` block.